lib/eldap/src/eldap.erl

```diff
@@ -12,30 +12,53 @@ References:
 
 The above publications can be found at [IETF](http://www.ietf.org).
 
-## DATA TYPES
+""".
 
-Type definitions that are used more than once in this module:
+-type connection_info() :: #{socket := ssl:sslsocket() | gen_tcp:socket(), socket_type := tcp | ssl}.
+-type search_option() ::
+    {base, string()} |
+    {filter, filter()} |
+    {scope, scope()} |
+    {attributes, [string()]} |
+    {deref, dereference()} |
+    {types_only, boolean()} |
+    {timeout, integer()}.
 
-- **`handle()`** - Connection handle
+-doc "An opaque handle unique for the connection".
+-type handle() :: term().
 
-- **`attribute() =`** - `{Type = string(), Values=[string()]}`
+-type attribute() :: {Type :: string(), Value :: [string()]}.
 
-- **`modify_op()`** - See `mod_add/2`, `mod_delete/2`, `mod_replace/2`
+-doc "See `mod_add/2`, `mod_delete/2`, `mod_replace/2`".
```

Comment:
I think we should have a description too,  not only a see refernce!

Reply:
let me give it a try to fix it.